artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -612,26 +640,15 @@ protected void onDeletePage(Page deletedPage) throws Exception {
    }
 
    /**
+    * This method will store the last known position for paging right before we stopped paging.
     * @param cursorList
     * @param currentPage
     * @throws Exception
     */
    protected void storeBookmark(ArrayList<PageSubscription> cursorList, Page currentPage) throws Exception {
-      try {
-         // First step: Move every cursor to the next bookmarked page (that was just created)
-         for (PageSubscription cursor : cursorList) {
-            cursor.confirmPosition(new PagePositionImpl(currentPage.getPageId(), -1));
-         }
-
-         // we just need to make sure the storage is done..
-         // if the thread pool is full, we will just log it once instead of looping
-         if (!storageManager.waitOnOperations(5000)) {
-            ActiveMQServerLogger.LOGGER.problemCompletingOperations(storageManager.getContext());
-         }
-      } finally {
-         for (PageSubscription cursor : cursorList) {
-            cursor.enableAutoCleanup();
-         }
+      // Move every cursor to the next bookmarked page
```

Comment:
how now are we making sure storage is done?

Reply:
I'm setting on a callback on the storage now.